src/agent/src/rpc.rs

```diff
@@ -236,9 +249,18 @@ impl AgentService {
 
         // start oom event loop
         if sid != cid && ctr.cgroup_manager.is_some() {
-            let cg_path = ctr.cgroup_manager.as_ref().unwrap().get_cg_path("memory");
+            let cg_path = ctr
+                .cgroup_manager
+                .as_ref()
+                .ok_or_else(|| anyhow!("cannot get cgroup manager"))?
+                .get_cg_path("memory");
+
             if cg_path.is_some() {
-                let rx = notifier::notify_oom(cid.as_str(), cg_path.unwrap()).await?;
+                let rx = notifier::notify_oom(
+                    cid.as_str(),
+                    cg_path.ok_or_else(|| anyhow!("cannot get cgroup path"))?,
+                )
+                .await?;
```

Comment:
Same for both these cases, we should use an `if let` instead of `if XXX.is_some() { XXX.unwrap()`.